lib/Doctrine/Migrations/Configuration/Loader/JsonFileLoader.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Configuration\Loader;
+
+use Doctrine\Migrations\Configuration\Configuration;
+use Doctrine\Migrations\Configuration\Exception\FileNotFound;
+use Doctrine\Migrations\Configuration\Exception\JsonNotValid;
+use const JSON_ERROR_NONE;
+use function assert;
+use function file_exists;
+use function file_get_contents;
+use function json_decode;
+use function json_last_error;
+
+/**
+ * @internal
+ */
+final class JsonFileLoader extends AbstractFileLoader
+{
+    /** @var ArrayLoader */
+    private $arrayLoader;
+
+    public function __construct(?ArrayLoader $arrayLoader = null)
```

Comment:
It's obviously a hard dependency, what do you think about making that clear by removing both the

- nullable type declaration and the
- default value

?

Reply:
set as hard dependency